django_app/redbox_app/redbox_core/models.py

```diff
@@ -747,6 +747,20 @@ def save(self, force_insert=False, force_update=False, using=None, update_fields
         super().save(force_insert, force_update, using, update_fields)
 
 
+class ExternalCitation(UUIDPrimaryKeyBase, TimeStampedModel):
```

Comment:
@jamesrichards4 im thinking that there are some big advantages to just extending the existing Citation model, what do you think?

```python
class Citation(UUIDPrimaryKeyBase, TimeStampedModel):
    class Origins(models.TextChoices):
        USER_FILES = "USER-FILES", _("User Files")
        WIKIPEDIA = "WIKIPEDIA", _("Wikpedia")
        GOV_UK = "GOV.UK", _("gov.uk")

    chat_message = models.ForeignKey("ChatMessage", on_delete=models.CASCADE)
    text = models.TextField()
    origin = models.TextField() # used to be called creator
    url = models.URLField(null=True, blank=True)
    file = models.ForeignKey(File, on_delete=models.CASCADE, null=True, blank=True)
    detail = JSONField(null=True, blank=True, help_text="could be page-number or uri fragment")
    origin = models.CharField(max_length=32, choices=Origins)

    def uri(self):
        return self.url or self.file.s3_url

    def save(self, ...):
        instance = super().save(...)
        if not instance.url and not instance.file:
            raise ValueError("either one of url or file must be set")
        return instance
```

Reply:
I'm also thinking the same. There's enough to distinguish between internal/external citations on the frontend (should we need to), e.g. `page_numbers` vs `url`.

Reply:
also, why dont we break this PR up into 2 bits, the redbox-core bit and the django bit?

Reply:
@KevinEtchells im thinking that the `origin` field should be the explicit internal/external source-of-truth? (in principle a well formed `uri` should be sufficient, in practice i dont think such things really exist!)

Reply:
How would this differ from the URL field?

Reply:
uri can begin with `file://` (and other schemes such as s3://) whereas url must begin with `http(s)://`

Reply:
I like the idea of a single model if we're confident it will provide the right details to break out the differences, the origin field feels like it does that to me but it's hard to be sure.

This should absolutely all have been broken down more, but now that it's here is it simpler to go ahead as is? I can break it out if you'd prefer though?

Reply:
sry, just seen your reply James, ive addd a PR with my suggested change here https://github.com/i-dot-ai/redbox/pull/1119 i can merge into yours?